### ISSUE TITLE
WebBrowser URI encoding bug

### ISSUE THREAD

**Issue by edi** [expo/expo#22202]
### Minimal reproducible example

https://snack.expo.dev/@swup/webbrowser-encoding-bug

### Summary

If you open my example and press both texts (`Linking.openURL` and `WebBrowser.openBrowserAsync`), it works just fine. However, if you run it on SDK 48 (must be done locally since `snack.expo.dev` doesn't support SDK 48 yet), the `WebBrowser` window opens a malformed URI (replacing all the intended `%2F` with slashes `/`), which leads to an error being returned by the site.

So, the URL is [https://firebasestorage.googleapis.com/v0/b/ride-starter.appspot.com/o/tickets%2FkvQwrhvk6k2odJw07nOG%2FxORfKEWH5rR6L4RwmyIA.jpg?alt=media&token=42c1d680-1680-4469-a296-cbc32304d9a9](https://firebasestorage.googleapis.com/v0/b/ride-starter.appspot.com/o/tickets%2FkvQwrhvk6k2odJw07nOG%2FxORfKEWH5rR6L4RwmyIA.jpg?alt=media&token=42c1d680-1680-4469-a296-cbc32304d9a9)

Note the use of `%2F` as directory separator, starting after`/o/tickets`, which is simply how `@google-cloud/storage` returns public URLs, and is a valid URL.

```javascript
let uri = 'https://firebasestorage.googleapis.com/v0/b/ride-starter.appspot.com/o/tickets%2FkvQwrhvk6k2odJw07nOG%2FxORfKEWH5rR6L4RwmyIA.jpg?alt=media&token=42c1d680-1680-4469-a296-cbc32304d9a9'

WebBrowser.openBrowserAsync(uri) // => decodes %2F into slashes and therefore breaks the URL.

// this is what WebBrowser ends up opening in the end, as you can press "open in browser (bottom right corner)" to see how the URL looks like.

// https://firebasestorage.googleapis.com/v0/b/ride-starter.appspot.com/o/tickets/kvQwrhvk6k2odJw07nOG/xORfKEWH5rR6L4RwmyIA.jpg?alt=media&token=42c1d680-1680-4469-a296-cbc32304d9a9
```

Please advise

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.3.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 16.18.1 - ~/.nvm/versions/node/v16.18.1/bin/node
      npm: 8.19.2 - ~/.nvm/versions/node/v16.18.1/bin/npm
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    IDEs:
      Xcode: 14.3/14E222b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~48.0.11 => 48.0.11 
      react: 18.2.0 => 18.2.0 
      react-native: 0.71.6 => 0.71.6 
    npmGlobalPackages:
      eas-cli: 3.8.1
      expo-cli: 6.3.2
    Expo Workflow: managed
```

**Comment by Tyson39** [comment]
1. - ### # ### No work ___****___

**Comment by Kudo** [comment]
hi @edi, i can reproduce this issue on ios expo go but not eas build. is it also your case?

**Comment by edi** [comment]
@Kudo I can confirm behaviour is only present in Expo Go client. Not sure about expo-dev-client tho, as I'm not currently using it anymore.

**Comment by Kudo** [comment]
thanks @edi. i'll follow up the investigation and update here.

**Comment by Kudo** [comment]
hi there! we've published ios expo go v2.28.9 including the fix. please try to update the latest expo go and that should resolve the problem for you. thanks!
